lib/rules/no-extra-parens.js

```diff
@@ -386,6 +386,30 @@ module.exports = {
             return node && (node.type === "Identifier" || node.type === "MemberExpression");
         }
 
+        /**
+         * Checks if a node is fixable.
+         * A node is fixable if removing a single pair of surrounding parentheses does not turn it
+         * into a directive after fixing other nodes.
+         * Almost all nodes are fixable, except if all of the following conditions are met:
+         * The node is a string Literal
+         * It has a single pair of parentheses
+         * It is the only child of an ExpressionStatement
+         * @param {ASTNode} node The node to evaluate.
+         * @returns {boolean} Whether or not the node is fixable.
+         * @private
+         */
+        function isFixable(node) {
+
+            // if it's not a string literal it can be autofixed
+            if (node.type !== "Literal" || typeof node.value !== "string") {
+                return true;
+            }
+            if (isParenthesisedTwice(node)) {
+                return true;
+            }
+            return node.parent.type !== "ExpressionStatement";
```

Comment:
Shall we also check if the `ExpressionStatement` is at the top level of a program or function body?

Reply:
I think yes, if it isn't at the top level of a program or function body then the fix seems safe.

Reply:
> I think yes, if it isn't at the top level of a program or function body then the fix seems safe.

Thanks! This is done in https://github.com/eslint/eslint/pull/17022/commits/7a2800d663b5fe5633c2f807c5a856796e6c2496. I also took the time to refactor similar code in other rules that do the same kind of checks.